crates/sui-core/src/execution_cache.rs

```diff
@@ -303,6 +307,8 @@ pub trait ObjectCacheRead: Send + Sync {
                 input_key
             );
             // If the key exists at the specified version, then the object is available.
+            // TODO-DNS does this need to be inverted to check markers first, so that it doesn't incorrectly load
```

Comment:
DNS comment

Reply:
and i think the answer is yes, clearly

Reply:
noting for posterity per offline discussion - this function is correct as-is, and when implementing execution logic for consensusv2 we'll need to add code to check after the object is loaded that the object owner is what we expect based on tx input args